### PR TITLE
chore: migrate to node:test with c8

### PR DESCRIPTION
**Pull request by narendra-reddy-nf**
This PR replaces tap with node:test and introduces c8 for code coverage.

# Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "codecov": "codecov",
     "lint": "standard | snazzy",
     "test": "npm run test:unit && npm run test:typescript",
-    "test:unit": "tap",
+    "test:unit": "c8 node --test",
```
Comment (gurgunday) [inline package.json:11]:
```suggestion
    "test:unit": "c8 --100 node --test",
```

File: package.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
     "qs": "^6.12.0",
     "snazzy": "^9.0.0",
     "standard": "^17.1.0",
-    "tap": "^18.7.1",
+    "c8": "^7.12.0",
```
Comment (gurgunday) [inline package.json:40]: why v7?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Uzlopak — CHANGES_REQUESTED** [review]
t.plan exists on node 20 and 22
https://nodejs.org/api/test.html#contextplancount

File: test/integration.test.js [hunk 3]
```diff
@@ -1,146 +1,156 @@
...
-const test = tap.test
+const test = require('node:test')
 const Fastify = require('fastify')
 const plugin = require('../')
 const qs = require('qs')
 const formAutoContent = require('form-auto-content')
 
-test('succes route succeeds', (t) => {
+test('success route succeeds', async (t) => {
   t.plan(3)
```
Comment (simoneb) [inline test/integration.test.js:10]: with the change that you did below, using t.plan is no longer necessary

File: test/integration.test.js [hunk 4]
```diff
@@ -1,146 +1,156 @@
...
+    ...formAutoContent({ foo: payload })
   })
+  t.assert.ifError(response.error)
+  t.assert.strictEqual(response.statusCode, 413)
+  t.assert.strictEqual(JSON.parse(response.body).message, 'Request body is too large')
 })
 
-test('cannot exceed body limit when Content-Length is not available', (t) => {
+test('cannot exceed body limit when Content-Length is not available', async (t) => {
   t.plan(3)
```
Comment (simoneb) [inline test/integration.test.js:55]: same as before

**Review by simoneb — CHANGES_REQUESTED** [review]

**Review by simoneb — APPROVED** [review]

**Review by gurgunday — APPROVED** [review]
lgtm

**Pull request merged**